var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -63,7 +63,8 @@ class Cmake(Package):
     depends_on('qt',             when='+qt')
     depends_on('python@2.7.11:', when='+doc', type='build')
     depends_on('py-sphinx',      when='+doc', type='build')
-    depends_on('openssl',        when='+openssl')
+    depends_on("openssl@:1.0.99", when='@:3.6.9+openssl')
+    depends_on("openssl", when='@3.7:+openssl')
```

Comment:
You should change this to:
```
depends_on("openssl", when='+openssl')
depends_on("openssl@:1.0.99", when='@:3.6.9+openssl')
```
Otherwise, with your changes:
```
$ spack spec cmake +openssl ^openssl@1.1.0d
Input spec
--------------------------------
cmake+openssl
    ^openssl@1.1.0d

Normalized
--------------------------------
==> Error: cmake does not depend on openssl
```
This is a bug in the concretizer, not really your fault. But there's no reason to specify `@3.7:` here since `cmake+openssl` always depends on `openssl`.